lib/pos_consensus.go

```diff
@@ -109,9 +109,24 @@ func (fc *FastHotStuffConsensus) Start() error {
 		time.Duration(currentSnapshotGlobalParams.BlockProductionIntervalMillisecondsPoS)
 	timeoutBaseDuration := time.Millisecond * time.Duration(currentSnapshotGlobalParams.TimeoutIntervalMillisecondsPoS)
 
+	// Refresh the checkpoint block info, so we can get tha latest view.
+	fc.blockchain.updateCheckpointBlockInfo()
+	checkpointBlockInfo := fc.blockchain.GetCheckpointBlockInfo()
+	currentView := uint64(0)
+	if checkpointBlockInfo != nil {
+		currentView = checkpointBlockInfo.LatestView
+	}
+
```

Comment:
You can do the view comparison here so it's encapsulated here where we perform the checkpoint request
```suggestion
	currentView := tipBlock.Header.GetView() + 1
	if checkpointBlockInfo != nil && checkpointBlockInfo.LatestView > currentView {
		currentView = checkpointBlockInfo.LatestView
	}
```